src/modules/ui/organisms/AddConfigurationForm/AddCredentialsForm.tsx

```diff
@@ -0,0 +1,179 @@
+  } = formMethods;
+
+  const { data: klarnaConfigurationData } =
+    trpcClient.paymentAppConfigurationRouter.paymentConfig.get.useQuery(
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- query is not enabled if configurationId is missing
+      { configurationId: configurationId! },
+      {
+        enabled: !!configurationId,
+        onError: (err) => {
+          getErrorHandler({
+            appBridge,
+            actionId,
+            message: "Error while fetching initial form data",
+            title: "Form error",
+          })(err);
+        },
+      },
+    );
+
+  useEffect(() => {
+    if (klarnaConfigurationData) {
+      reset({
+        apiUrl: klarnaConfigurationData.apiUrl,
+        username: klarnaConfigurationData.username,
+        password: klarnaConfigurationData.password,
+        configurationName: klarnaConfigurationData.configurationName,
+      });
+    }
+  }, [klarnaConfigurationData, reset]);
+
+  const getOnSuccess = useCallback(
+    (message: string) => {
+      return () => {
+        void appBridge?.dispatch({
+          type: "notification",
+          payload: {
+            title: "Form saved",
+            text: message,
+            status: "success",
+            actionId,
+          },
+        });
+        void context.paymentAppConfigurationRouter.paymentConfig.invalidate();
+      };
+    },
+    [appBridge, context.paymentAppConfigurationRouter.paymentConfig],
+  );
+
+  const onError = getFieldErrorHandler({
+    appBridge,
+    setError,
+    actionId,
+    fieldName: "root",
+    formFields: getFormFields(defaultValues),
+  });
+
+  const { mutate: updateConfig } =
+    trpcClient.paymentAppConfigurationRouter.paymentConfig.update.useMutation({
+      onSuccess: (data) => {
+        invariant(data.configurationId);
+        getOnSuccess("App configuration was updated successfully")();
+        return router.replace(`/configurations/edit/${data.configurationId}`);
+      },
+      onError,
+    });
+  const { mutate: addNewConfig } =
+    trpcClient.paymentAppConfigurationRouter.paymentConfig.add.useMutation({
+      onSuccess: async (data) => {
+        invariant(data.configurationId);
+        context.paymentAppConfigurationRouter.paymentConfig.get.setData(
+          { configurationId: data.configurationId },
+          data,
+        );
+        if (!configurationId) {
+          await router.replace(`/configurations/edit/${data.configurationId}`);
+        }
+      },
+      onError,
+    });
+
+  const handleConfigSave: SubmitHandler<PaymentAppFormConfigEntry> = (data) => {
+    configurationId
+      ? updateConfig({
+          configurationId,
+          entry: data,
+        })
+      : addNewConfig(data);
+  };
+
+  return (
+    <RoundedBoxWithFooter
+      as="form"
+      method="POST"
+      autoComplete="off"
+      // eslint-disable-next-line @typescript-eslint/no-misused-promises
+      onSubmit={handleSubmit(handleConfigSave, console.warn)}
```

Comment:
Will `console.warn` override the error handler from `useForm`? I think we should also use `Sentry.captureException` or `Sentry.captureMessage` here so that we can track errors